app/experimenter/nimbus-ui/src/components/PageEditBranches/FormBranches/index.tsx

```diff
@@ -97,9 +97,26 @@ export const FormBranches = ({
 
   const clearSubmitErrors = () => dispatch({ type: "clearSubmitErrors" });
 
+  // We show two branches by default, an empty control/reference branch and treatment
+  // branch, which are given a default branch name. If a branch slug is not set, it means
+  // that branch hasn't been saved by the user. If only a reference branch was previously
+  // saved we _don't_ show a new treatment branch by default and instead, if "handleAddBranch"
+  // is called, we show the empty treatment branch instead of creating a new one.
```

Comment:
This is a pretty beefy comment but I thought this was kind of hard to look at otherwise and could use an explainer, especially after not looking at branches in a while and coming back to it.

Reply:
Yeah it's a bit hairy but it all works right from my local testing so good to have the explanation 👍